languages/rust/template/src/bin/pipe.rs

```diff
@@ -0,0 +1,165 @@
+// This file provides interop with the langserver
```

Comment:
one comment on this being in the `template` is that it means algorithm developers can muck with it since it would be in the algorithm's repo (and also it makes it way harder to update across all users).  since this is all statically built i get that the file needs to be available but it could also live in `libraries/rust-buildtime/context/pipe.rs` and copied into the builder image at a well-known location like `/opt/algorithmiaio/rust/pipe.rs` or whatever if there's a way to include that file as part of the build script (or have one line in the build script be `cp /opt/algorithmia/rust/pipe.rs /opt/algorithm/src/pipe.rs` etc).  definitely assists in some ways but would potentially silently overwrite a file with the name pipe.rs if a user actually made that in an algorithm.  idk anything about rust so don't know how hard it is to include sources from multiple locations

Reply:
I have removed `pipe.rs`. You will now see the former `lib.rs` became `main.rs` with a very short `fn main()` that registers the apply function. I'd love feedback on this new approach (even open to bikeshedding the precise names, e.g. `setup_handler`)